lib/Service/DocumentService.php

```diff
@@ -318,7 +322,11 @@ public function autosave($file, $documentId, $version, $autoaveDocument, $force
 		}
 		$this->cache->set('document-save-lock-' . $documentId, true, 10);
 		try {
-			$file->putContent($autoaveDocument);
+			if (empty($autoaveDocument)) {
+				$file->delete();
```

Comment:
This will not work, if the file gets deleted the file id will be invalid and therefore the editing session that users might be currently connected to will fail.

Reply:
I'm not sure if there is an easy way to solve this other than introducing a more decoupled session handling for rich workspaces that will only use the target file for storing the results, however we don't have a unique identifier then for concurrent editing sessions anymore (as this is what the file is used for). If a file gets deleted and recreated it will get a new file id.

Reply:
Well, in this case (the user did not enter anything/removed content) we want the file to be deleted.
If it were deleted in the meantime however, that should be fine. So do we simply need to handle DoesNotExistExceptions here?